#100 opencl jordan updates

jordan-heemskerk opened the pull request.

jordan-heemskerk committed:
dynfu::kinect_fusion_opencl_pose_estimation_pipeline_block

 - Fix logic error in threshold on number of rejected
   correspondences.

jordan-heemskerk committed:
client: add --max-frames option

jordan-heemskerk committed:
dynfu::msrc_file_system_depth_device: update K

jordan-heemskerk committed:
client: Add TSDF viewer when seq is complete

jordan-heemskerk committed:
dynfu::kinect_fusion_opencl_update_reconstruction_pipeline_block

 - Instead of generating a TSDF, just update using an average

jordan-heemskerk committed:
dynfu:: kinect_fusion_opencl_update_reconstruction_pipeline_block

 - Decrease default tsdf resolution for debugging so that rendering
   is faster

RobertLeahy commented on src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp:
```diff
@@ -165,8 +165,8 @@ namespace dynfu {
 
-		std::uint32_t threshold(frame_height_*frame_width_);
-		threshold/=10U;
+		std::uint32_t threshold (frame_height_*frame_width_*0.9f);// can have up to 90% rejected
```
I'm not the world's biggest fan of this implicit to `float` and implicit to `std::uint32_t`, especially since at very high values `std::uint32_t` has much greater precision in terms of integer values than `float`.

Consider:

```
std::uint32_t threshold(frame_height_*frame_width_);
threshold/=10U;
threshold*=9U;
```

Also: We can probably hoist this calculation out of the loop.

RobertLeahy committed:
dynfu::kinect_fusion_eigen_pose_estimation_pipeline_block & dynfu::kinect_fusion_opencl_pose_estimation_pipeline_block - Clang Fixes

Wouldn't build on Mac/Clang due to an issue with std::unique_ptr constructors.

RobertLeahy committed:
CMakeLists.txt - Mac/Linux Fixes

RobertLeahy committed:
CMakeLists.txt - Mac/Linux Fixes

Need to link "client" against GLFW.

jordan-heemskerk committed:
tsdf_kernel: Write a NaN if no measurement

RobertLeahy committed:
dynfu::libigl & Link Time Fixes

Created the dynfu::libigl namespace which contains extremely thin wrappers to libigl functions.  dynfu::libigl is linked into a separate shared library which allows it to be built/linked once and never touched again.

This greatly reduces link times for "client" and "tsdf_viewer".

RobertLeahy committed:
Merge pull request #134 from RobertLeahy/#133-link

dynfu::libigl & Link Time Fixes

jordan-heemskerk committed:
tsdf.cl: Write out a NaN if first iteration

jordan-heemskerk committed:
client: inform user of overall frame processing progress

jordan-heemskerk committed:
client: Remove counting of NaNs, just set them =1

RobertLeahy commented on src/msrc_file_system_depth_device.cpp:
```diff
@@ -73,7 +73,7 @@ namespace dynfu {
 		Eigen::Matrix3f k;
 		k << 585.0f, 0.0f, 320.0f,
-		     0.0f, -585.0f, 240.0f,
+		     0.0f, 585.0f, 240.0f,
```
What's the deal with this change?

jordan-heemskerk replied:
Focal length should be positive. That was a previous mistake. It flips the frame upside down in Y.
